### PR TITLE
Initial public draft of governance doc

### PR DESCRIPTION
**Pull request by Mierdin**
This is the initial draft for the NRE Labs governance document. Please use the "files changed" tab to inspect the contents and provide in-line comments where relevant.

The credit for this draft goes to @LCaywood - she asked that I post this via PR on her behalf.

### CODE REVIEW

File: governance.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+#### Voting Procedures
+
+- 1 vote per Steering Committee Voter
+- Other voting members may be admitted by majority vote of existing Voters.
+- If a Steering Committee Voter misses 3 meetings in a row without designating a proxy, their voting capacity will be suspended until they have attended 2 meetings in succession.
+- If a Steering Committee Voter misses 4 or more meetings in a row, with or without a proxy, the Steering Committee may remove that individual and replace them with someone with greater availability.
+
+Elections and any other voting decisions require a quorum of 50% of voting members. Voting can be held during Committee meetings, or by electronic means within a specified time period.
+
+Amendments to this or Policy documents require a majority vote.
```
Comment (LCaywood) [inline governance.md]: ...of the Steering Committee.

File: governance.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+- Regularly attend Technical Community meetings and work closely with the Technical Community Leader to ensure ongoing alignment between the Steering Committee and Technical Community.
+
+The Treasurer will be elected according to the same criteria and process as the Director. The Treasurer will have the following responsibilities.
+
+- Advise on and validate operating plan’s budget.
+- Review payments authorized by the Director each month, and report regularly on revenues and expenditures in relation to budget.
+
+The Marketing Lead will be elected according to the same criteria and process as the Director. The Marketing Lead will have the following responsibilities.
+
+- Develop the Marketing portion, including metris, of the annual operating plan in consultation with the Technical Community. Report regularly to the Steering Committee on progress against plan.
```
Comment (smk4664) [inline governance.md]: "including metris" should be "including metrics".

Comment (LCaywood) [inline governance.md]: metrics, not metris

File: governance.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+Elections and any other voting decisions require a quorum of 50% of voting members. Voting can be held during Committee meetings, or by electronic means within a specified time period.
+
+Amendments to this or Policy documents require a majority vote of the Steering Committee.
+
+### Compliance with Policies and Licenses
+
+Contributors will comply with the policies adopted by the Steering Committee of NRE Labs. These policies may include:
+
+- Contributor License Agreement
+- Code of Conduct
```
Comment (olberger) [inline governance.md]: TODO: Add link to the CoC doc

File: governance.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+- Privacy Policy
+- Terms of Use
+- Trademark and Branding Policy
+
+#### Community Assets and Intellectual Property 
+
+Title to all trade or service marks used by the Project (“Project Trademarks”) is currently held by Juniper Networks. In the likely event that the Project joins an open source foundation or fiscal conservancy, Project Trademarks shall be transferred and assigned to that entity to hold on behalf of the Project. Any use of any Project Trademarks by participants in the Project or others shall be in accordance with the Trademark and Branding Policy.
+
+The Project shall develop and own all Project repositories and social media accounts, and domain name registrations created by the Project. The Platform is licensed under the Apache v2.0 license. Lessons contributed to the Project are published under CC BY 2.0. Project documentation, web and marketing content shall be published under Creative Commons license.
```
Comment (olberger) [inline governance.md]: I'd prefer to see some clear Copyright + License heading rather than a sentence inside that general section: these are probably the most important legal elements that contributors or users (reusers) should be concerned about IMHO (like, "can I copy/paste it to reuse it in class ?", or "can I fork the Antidote/Syringe Go code to create another SaaS startup ?"...)

File: governance.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
+## Mission & Scope
```
Comment (olberger) [inline governance.md:1]: I'd welcome an introductory section to describe the purpose of that very document, and who should be concerned by reading it: mission of the document vs mission of the project ;-)

File: governance.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+NRE Labs grew out of the experiences of working network engineers who were quick to see the value of embracing infrastructure automation for the purpose of network reliability, but were challenged by the learning curve of doing so.
```
Comment (olberger) [inline governance.md:5]: TODO: Expand the NRE acronym ?

File: governance.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
+## Mission & Scope
+
```
Comment (olberger) [inline governance.md:2]: Maybe a subheading "History", or similar here : or "Where did it came from". This one is rather opinionated and not likely to change, whereas the "mission" later may evolve in the future ?

File: governance.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+### Mission
+
+NRE Labs has been an open source project from its inception--both the platform itself and all the lessons. The NRE Labs community seeks to ensure that anyone anywhere can quickly learn key automation concepts using whatever network equipment and tools they have available. Thus we encourage contributions from as broad a swath of the networking and automation communities as possible.  Anyone can contribute to any aspect of the project, according to the rules laid out in this governance document.
+
+We resolve to provide a platform for education in infrastructure automation tools and techniques for network engineers that can immediately be applied to real-world daily tasks.
+
+We encourage lessons of all types relating to network reliability engineering and infrastructure automation. All lessons should be concise and practical, focused on the mastery of a specific automation task.
+
```
Comment (olberger) [inline governance.md:18]: Question : should the site/lessons limit to education about publicly available (i.e. open source) tools and techniques or should it be broader (incl. proprietary products) ?

File: governance.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+The group of engineers who started NRE Labs believes strongly that expecting an entire profession to start over and learn programming from first principles is most likely to result in a general failure to adopt network automation at all, even as networking itself becomes ever more important in the basic functioning of distributed, multicloud operations.
+
+### Mission
+
+NRE Labs has been an open source project from its inception--both the platform itself and all the lessons. The NRE Labs community seeks to ensure that anyone anywhere can quickly learn key automation concepts using whatever network equipment and tools they have available. Thus we encourage contributions from as broad a swath of the networking and automation communities as possible.  Anyone can contribute to any aspect of the project, according to the rules laid out in this governance document.
```
Comment (olberger) [inline governance.md]: The last sentence is a bit problematic : anyone could contribute to that governance document in principle, but not as easimy as contributing a lesson. One needs to climb a few steps inside the community in between. While the intent is valuable, I feel it's a bit too extensive in reality ;-)

File: governance.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+We resolve to provide a platform for education in infrastructure automation tools and techniques for network engineers that can immediately be applied to real-world daily tasks.
+
+We encourage lessons of all types relating to network reliability engineering and infrastructure automation. All lessons should be concise and practical, focused on the mastery of a specific automation task.
+
+### Scope
+
+The scope of the Project includes software development under an OSI-approved open source licence supporting the mission, including documentation, testing, integration and the creation of other artifacts that aid the development, deployment, operation or adoption of the open source software project.
```
Comment (olberger) [inline governance.md]: That first sentence seems rather a template from a generic document: may need to be adapted: "The scope of NRE Labs...", and the repetition of " the open source software project" at the end is a bit redundant : just "the project" : lessons aren't typically open source software in some sense (hence different licences, etc.) so... not a big deal anyway.

File: governance.md [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+
+NRE Labs has two main components: the Platform (“Antidote”), and the Lessons. The Platform can serve a wide variety of purposes as a lesson delivery platform, and similarly, Lessons may be written for a wide variety of subjects which do not directly pertain to the Mission of the project.
```
Comment (olberger) [inline governance.md:23]: **currently** has two main components.

Maybe make it explicit here that the Platform is open source software, to make it explicit that you may wish to operate your own Antidote if you wish, and still be welcome in the Project ?

Comment (olberger) [reply governance.md:23]: In general, I fear the articulation between the 2 main components is something that's the most complex originality here : thinking about the codebase of Wordpress vs wordpress.com for instance. What if I want to contribute to Antidote platform something that has no value to NRE Labs' curriculum... I fear some roadmap tension here...

Not sure a separate governance for the Platform software, and another for the site's curriculum would be easier.

I'm just playing the devil's advocate, and overengineering ;-)

File: governance.md [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+Technical participation in the Project is open to anyone so long as they abide by the terms of this document and documents referenced herein. Technical Community meetings are conducted in English and are open to the public and recorded. The Technical Community generally will involve Contributors and Committers. Contributors include anyone in the Technical Community who contributes code, documentation, lessons or other artifacts to the Project. Committers are Contributors who have earned the ability to modify (“commit”) source code, documentation or other artifacts in a project’s repository. Each Project will ideally have at least 2 Committers.
```
Comment (olberger) [inline governance.md]: Not sure recording of the TC meetings should be mandated upfront in this doc. Documented at least, maybe ? Yay for transparency, but should all verbatims be systematically made available ?

File: governance.md [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+
+While the Project aims to operate as a consensus-based community, if any Technical Community decision requires a vote to move the Project forward, the eligible voters in the Technical Community will do so by simple majority. Votes may be timebound and held either during meetings or electronically, with the voters’ choices publicly recorded, but at least 50% of eligible Community voters must participate within the specified period of time in order for a vote to take effect. With the exception of changes to Governance or Policy, votes will be decided by a simple majority or greatest plurality (in the case of multiple options). Governance or Policy changes will require approval of ⅔ of the voting quorum.
+
+### Steering Committee
+
+The Steering Committee is responsible for setting the strategic direction of the Project in accordance with its Mission. It provides approval and oversight of the Project’s budget and disbursements, Policy documents and other legal matters, marketing, and other non-technical activities of the Project. Committee business will be conducted in English and will be recorded for public access.
```
Comment (olberger) [inline governance.md]: Again, recording (as in Youtube videos) seems a bit too strong a requirement : discussing confidential matters like potential sponsors, exclusion of people, etc. Such should probably be conducted in private channels until agreed on, for legal reasons, so I doubt recordings could be made, in practice. I'd suggest to only mandate documentation/minutes of the meetings, allowing to exclude confidential elements from the records.

File: governance.md [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+
+Elections and any other voting decisions require a quorum of 50% of voting members. Voting can be held during Committee meetings, or by electronic means within a specified time period.
+
+Amendments to this or Policy documents require a majority vote of the Steering Committee.
+
+### Compliance with Policies and Licenses
+
+Contributors will comply with the policies adopted by the Steering Committee of NRE Labs. These policies may include:
+
+- Contributor License Agreement
```
Comment (olberger) [inline governance.md]: Is there a proposed template for that one ? Does that mean that the Project should hold the copyright for all contributions, then ?

File: governance.md [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+- Code of Conduct
+- Privacy Policy
```
Comment (olberger) [inline governance.md]: Is this related to Operations of the platform only ?

File: governance.md [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+- Terms of Use
```
Comment (olberger) [inline governance.md]: use of the platform as someone learning network automation, or use of the resources of the Project for committers ?

File: governance.md (hunk 8 above)
Comment (LCaywood) [reply governance.md:18]: FAQ - clarify in contribution docs

File: governance.md (hunk 15 above)
Comment (LCaywood) [reply governance.md]: Also relates to the operations of the community and to the privacy of the community. Need to confirm how Discourse & Discord's privacy policies would operate under a general privacy policy (likely adopted from the conservancy/foundation the Project may eventually be hosted by).

File: governance.md (hunk 16 above)
Comment (LCaywood) [reply governance.md]: Both

File: governance.md (hunk 4 above)
Comment (LCaywood) [reply governance.md]: Will moved the license info further up in the doc. There may be a good call for an FAQ describing some of these things in more detail, but not needed in a governance doc as such.

File: governance.md (hunk 11 above)
Comment (LCaywood) [reply governance.md:23]: Not helpful in a governance doc, which discusses how the project is to be run, but something to be included in a contributor FAQ.

File: governance.md (hunk 14 above)
Comment (LCaywood) [reply governance.md]: We've been advised to have contributors use the DCO https://developercertificate.org/ vs the heavier CLA. This will be included in the contributor documentation.

File: governance.md (hunk 15 above)
Comment (LCaywood) [reply governance.md]: Legal advisors have suggested incorporating the Privacy Policy within the Terms of Use.

**Comment by Mierdin** [comment]
This was ratified on the weekly standup on October 22, 2019.
